webcomponents-loader.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+(function() {
+  // Feature detect which polyfill needs to be imported.
+  let polyfills = [];
+  let useNativeImports = ('import' in document.createElement('link'));
+  if (!useNativeImports) {
+    polyfills.push('hi');
+  }
+
+  // Stub out HTMLImports if we're using native imports
+  window.HTMLImports = {
+    whenReady: function(callback) {
+      if (useNativeImports) {
+        // When native imports boot, the are "ready" the first rAF after
+        // the document becomes interactive, so wait for the correct state change.
+        if (document.readyState !== 'interactive') {
+          function once() {
+            document.removeEventListener('readystatechange', once);
+            window.HTMLImports.whenReady(callback);
+          }
+          document.addEventListener('readystatechange', once);
+        } else {
+          requestAnimationFrame(function() {
+            callback();
```

Comment:
Ugh, fake-polyfilling an API that comes from a polyfill is a path down a dark alley, in my opinion. This is to fix the problem that a lot of code (including our friend `browser.js` relies on this callback being called), but if your use case is more advanced than that (and I don't see anyone using `importInfo`), then not sure we should help in that case.